speechbrain/nnet/losses.py

```diff
@@ -1128,3 +1128,47 @@ def nll_loss_kd(
     # Loss averaging
     loss = torch.sum(loss.reshape(N_snt, max_len) * mask) / torch.sum(mask)
     return loss
+
+
+class ContrastiveLoss(nn.Module):
```

Comment:
Where did you get your inspiration from ? If the code is taken from Fairseq or HF, we must mention it.

Reply:
will do

Reply:
done